hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/action/compact/CompactionTriggerStrategy.java

```diff
@@ -18,13 +18,116 @@
 
 package org.apache.hudi.table.action.compact;
 
+import java.text.ParseException;
+import org.apache.hudi.common.table.timeline.HoodieActiveTimeline;
+import org.apache.hudi.common.util.collection.Pair;
+import org.apache.hudi.exception.HoodieCompactionException;
+import org.apache.log4j.LogManager;
+import org.apache.log4j.Logger;
+
+@SuppressWarnings("checkstyle:LineLength")
 public enum CompactionTriggerStrategy {
-    // trigger compaction when reach N delta commits
-    NUM_COMMITS,
-    // trigger compaction when time elapsed > N seconds since last compaction
-    TIME_ELAPSED,
-    // trigger compaction when both NUM_COMMITS and TIME_ELAPSED are satisfied
-    NUM_AND_TIME,
-    // trigger compaction when NUM_COMMITS or TIME_ELAPSED is satisfied
-    NUM_OR_TIME
+
+  // trigger compaction when reach N delta commits
+  NUM_COMMITS {
+    @Override
+    public boolean compactable(
+        long compactInlineMaxDeltaCommits,
```

Comment:
can we wrap these around, without placing them on each line, consistent with most of our code?

Reply:
@vinothchandar  What about replacing `inlineCompactDeltaCommitMax, inlineCompactDeltaSecondsMax` with `HoodieWriteConfig config` ?